### PR TITLE
Fixes #16587: Rudder-cli does not generate correctly the API

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/16587

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

File: lib.python/generate.py [hunk 1]
```diff
@@ -67,7 +67,12 @@ def generate(function):
     print('    clean_params(data)')
   final_url = '"' + url.replace('{', '" + ').replace('}', ' + "') + '"'
   if method == 'GET':
-    print('    return self.request("' + method + '", ' + final_url + ', return_raw=return_raw)')
+    print('    extra=""')
+    for param in optional_params:
+      print("    if %s is not None:"%(param))
+      print('      extra += "?%s=" + %s'%(param, param))
```
Comment (peckpeck) [inline lib.python/generate.py]: I think you need to urlformat the params : https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote

**Comment by Fdall** [comment]
PR updated with a new commit

**Pull request merged**